### PR TITLE
JDK-8305100 [REDO] Clean up JavadocTokenizer

### PR DESCRIPTION
**Pull request by JimLaskey**
Previous work had cleaned up the JavacTokenizer, but left JavadocTokenizer alone, mostly due to its perplexing state. This is follow up work to simplify the JavadocTokenizer.

Redo changes:

1. Needed to check for end of data after @deprecated
2. Position of newline in offset map was incorrect (out by one).
3. Skip whitespace after first line (put offsets out by one).

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305100](https://bugs.openjdk.org/browse/JDK-8305100): [REDO] Clean up JavadocTokenizer

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13265/head:pull/13265` \
`$ git checkout pull/13265`

Update a local copy of the PR: \
`$ git checkout pull/13265` \
`$ git pull https://git.openjdk.org/jdk.git pull/13265/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13265`

View PR using the GUI difftool: \
`$ git pr show -t 13265`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13265.diff">https://git.openjdk.org/jdk/pull/13265.diff</a>

</details>

### CODE REVIEW

**Comment by JimLaskey** [comment]
tier1,tier2,tier3 mach5 run https://mach5.us.oracle.com/mdash/jobs/jlaskey-open-20230330-1940-43993922

**Comment by JimLaskey** [comment]
Original PR https://github.com/openjdk/jdk/pull/12953

File: src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavaTokenizer.java [hunk 1]
```diff
@@ -1273,76 +1255,110 @@ public CommentStyle getStyle() {
...
+                    int pos = line.position();
+                    line.skipWhitespace();
+
+                    if (line.accept("@deprecated") &&
+                            (!line.isAvailable() ||
+                                    line.isWhitespace() ||
+                                    line.isEOLN() ||
+                                    line.get() == EOI)) {
+                        deprecatedFlag = true;
+                    }
```
Comment (jonathan-gibbons) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavaTokenizer.java:1357]: As mentioned recently elsewhere, this is generally insufficient, albeit no worse than before.
Generally, we should move away from this code, and eliminate `deprecated Flag` entirely. Deprecation is (or should be) indicated by the `@Deprecated` annotation, with associated documentation given by a plain/regular `@deprecated` tag, as read by `DocCommentParser` or equivalent.

**Review by jonathan-gibbons — APPROVED** [review]
Looks good to me

**Pull request closed without merging**